### PR TITLE
test: Code Coverage: Delegated Accounts Page

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at 298633e</samp>

Updated the test file for the delegated accounts page to import the necessary modules and services, and to use fakeAsync and tick for testing asynchronous operations. Also added mock data and services to simulate the delegated accounts functionality.

### <samp>🤖 Generated by Copilot at 298633e</samp>

> _`fakeAsync`, `tick`_
> _More imports for testing_
> _Winter of code_

### <samp>🤖 Generated by Copilot at 298633e</samp>

*  Add imports for testing modules and services ([link](https://github.com/fylein/fyle-mobile-app/pull/2434/files?diff=unified&w=0#diff-7762b1fb56ad4054f9ab823ec45a6f029b2b3405de5087bd6b92480272a2bfe7L1-R172))

## Clickup
https://app.clickup.com/t/85ztznwc3

## Code Coverage
`100% Statements 38/38`
`100% Branches 11/11`
`100% Functions 20/20`
`100% Lines 37/37`

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/delegated-accounts/delegated-accounts.page.spec.ts [hunk 1]
```diff
@@ -1,26 +1,173 @@
...
+
+      expect(loaderService.showLoader).toHaveBeenCalledTimes(1);
+      expect(loaderService.hideLoader).toHaveBeenCalledTimes(1);
+      expect(recentLocalStorageItemsService.clearRecentLocalStorageCache).toHaveBeenCalledTimes(1);
+      expect(orgUserService.switchToDelegatee).toHaveBeenCalledTimes(1);
+      expect(router.navigate).toHaveBeenCalledOnceWith(['/', 'enterprise', 'my_dashboard']);
+    }));
+
+    it('should allow user to search and select a delegatee account', fakeAsync(() => {
+      component.searchDelegatees = fixture.debugElement.query(By.css('.delegated--search-input'));
```
Comment (OmkarJ13) [inline src/app/fyle/delegated-accounts/delegated-accounts.page.spec.ts:132]: Use utility method for this

File: src/app/fyle/delegated-accounts/delegated-accounts.page.spec.ts [hunk 2]
```diff
@@ -1,26 +1,173 @@
...
+      tick(500);
+
+      expect(component.delegatedAccList).toEqual([eouRes2, eouRes3]);
+      expect(orgUserService.findDelegatedAccounts).toHaveBeenCalledTimes(1);
+      expect(orgService.getCurrentOrg).toHaveBeenCalledTimes(1);
+      expect(orgUserService.excludeByStatus).toHaveBeenCalledWith([apiEouRes, eouRes2, eouRes3], 'DISABLED');
+    }));
+
+    it('should set delegatee acc list to empty array if no accounts are provided', fakeAsync(() => {
+      component.searchDelegatees = fixture.debugElement.query(By.css('.delegated--search-input'));
```
Comment (OmkarJ13) [inline src/app/fyle/delegated-accounts/delegated-accounts.page.spec.ts:153]: Same as above

**Review by OmkarJ13 — APPROVED** [review]

**Review by Chethan-Fyle — APPROVED** [review]

File: src/app/fyle/delegated-accounts/delegated-accounts.page.spec.ts (hunk 1 above)
Comment (Dimple16) [reply src/app/fyle/delegated-accounts/delegated-accounts.page.spec.ts:132]: +1

File: src/app/fyle/delegated-accounts/delegated-accounts.page.spec.ts (hunk 2 above)
Comment (Dimple16) [reply src/app/fyle/delegated-accounts/delegated-accounts.page.spec.ts:153]: +1

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

File: src/app/fyle/delegated-accounts/delegated-accounts.page.spec.ts [hunk 3]
```diff
@@ -1,26 +1,173 @@
...
+      const input = component.searchDelegatees.nativeElement as HTMLInputElement;
+      activatedRoute.snapshot.params.switchToOwn = null;
+      orgUserService.findDelegatedAccounts.and.returnValue(of([apiEouRes, eouRes2, eouRes3]));
+      orgService.getCurrentOrg.and.returnValue(of(orgData1[0]));
+      orgUserService.excludeByStatus.and.returnValue([eouRes2, eouRes3]);
+
+      component.ionViewWillEnter();
+      tick(500);
+
+      input.value = 'ajain@fyle.in';
```
Comment (Dimple16) [inline src/app/fyle/delegated-accounts/delegated-accounts.page.spec.ts:142]: check if we can add this in the beginning once and reuse it in other places, that'll avoid repetition

**Review by Dimple16 — APPROVED** [review]
one comment

**Pull request merged**